hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/codec/BurnWrapper.java

```diff
@@ -25,21 +25,21 @@
 import java.util.Collections;
 import java.util.List;
 
-public record BurnWrapper(long amount, TokenID tokenType, List<Long> serialNos) {
-    private static final long NONFUNGIBLE_BURN_AMOUNT = -1;
+public record BurnWrapper(long amount, TokenID tokenType, List<Long> serialNos, boolean fungibleToken) {
```

Comment:
Can we make the `long amount` to `Long amount` , so we can set amount to null for non fungible types. Instead of 0 or -1.